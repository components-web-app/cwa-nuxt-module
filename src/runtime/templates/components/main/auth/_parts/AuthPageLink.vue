<template>
  <div class="cwa:text-sm">
    <NuxtLink
      :to="props.linkTo"
      class="cwa:font-medium cwa:text-neutral-400 cwa:hover:text-neutral-200"
    >
      {{ props.linkText }}
    </NuxtLink>
  </div>
</template>

<script setup lang="ts">
const props = defineProps<{ linkText: string, linkTo: string }>()
</script>
