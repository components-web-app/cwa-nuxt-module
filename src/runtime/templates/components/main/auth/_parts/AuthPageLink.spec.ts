// @vitest-environment happy-dom
import { describe, expect, test } from 'vitest'
import { shallowMount } from '@vue/test-utils'
import AuthPageLink from './AuthPageLink.vue'

function createWrapper() {
  return shallowMount(AuthPageLink, {
    props: {
      linkText: 'Click here to get money',
      linkTo: '/bank/loan',
    },
  })
}

describe('AuthPageLink', () => {
  describe('snapshots', () => {
    test('should match snapshot', () => {
      const wrapper = createWrapper()

      expect(wrapper.element).toMatchSnapshot()
    })
  })
})
