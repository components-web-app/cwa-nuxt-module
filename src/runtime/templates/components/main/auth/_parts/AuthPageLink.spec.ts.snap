// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AuthPageLink > snapshots > should match snapshot 1`] = `
<div
  class="cwa:text-sm"
>
  <nuxtlink-stub
    class="cwa:font-medium cwa:text-neutral-400 cwa:hover:text-neutral-200"
    to="/bank/loan"
  >
    Click here to get money
  </nuxtlink-stub>
</div>
`;
